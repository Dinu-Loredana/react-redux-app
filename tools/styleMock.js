// Mocks CSS imports for Jest. Tells jest that sees us importing CSS, should ignore it.
module.exports = {};
