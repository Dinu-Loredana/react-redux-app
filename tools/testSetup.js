import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"; // configure Enzyme to work with React17
configure({ adapter: new Adapter() });

// install it via npm (npm i @wojtekmaj/enzyme-adapter-react-17).
// You will need to install enzyme along with an Adapter corresponding to the version of react (or other UI Component library) you are using (React 17).
// Add to package.json below Jest:
// "setupFiles":[
//  "./tools/testSetup.js"  (jest'll run this file)
// ],
