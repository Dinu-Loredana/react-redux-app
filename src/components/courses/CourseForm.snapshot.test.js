import React from "react";
import CourseForm from "./CourseForm";
import renderer from "react-test-renderer";
import { authors, courses } from "../../../tools/mockData";

// check if the label of Save button is properly set to 'Saving...' when saving state is true
it("sets submit label button to 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onChange={jest.fn()} // creates an empty mock function
      onSave={jest.fn()}
      errors={{}}
      saving //pass the props that the component receives, use mock data (same mock data is used for Mock API)
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit label button to 'Save' when saving is false", () => {
  const tree = renderer.create(
    <CourseForm
      course={courses[0]}
      authors={authors}
      onChange={jest.fn()}
      onSave={jest.fn()}
      errors={{}}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
