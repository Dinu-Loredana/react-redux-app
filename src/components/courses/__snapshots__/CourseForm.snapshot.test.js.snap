// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit label button to 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h6
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Edit
     Course
  </h6>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        data-testid="select"
        id="authorId"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option>
          Select Author
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        id="category"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit label button to 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h6
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Edit
     Course
  </h6>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        data-testid="select"
        id="authorId"
        name="authorId"
        onChange={[MockFunction]}
        value={1}
      >
        <option>
          Select Author
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        id="category"
        name="category"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
